services/streammanager/kafka/client/producer.go

```diff
@@ -109,17 +108,12 @@ func (p *Producer) Close(ctx context.Context) error {
 func (p *Producer) Publish(ctx context.Context, msgs ...Message) error {
 	messages := make([]kafka.Message, len(msgs))
 	for i := range msgs {
-		var headers []kafka.Header
-		if l := len(msgs[i].Headers); l > 0 {
-			headers = make([]kafka.Header, l)
-			for k := range msgs[i].Headers {
-				headers[k] = kafka.Header{
-					Key:   msgs[i].Headers[k].Key,
-					Value: msgs[i].Headers[k].Value,
-				}
-			}
+		if msgs[i].Topic == "" {
+			return fmt.Errorf("no topic provided for message %d", i)
```

Comment:
looks like we are missing coverage for this scenario

Reply:
I think codecov complains here because it is covered from another package. We don't have a test about this in `client_test.go` where it is its natural fit. It's OK anyway I guess since it is covered.